app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+        self.background.size.set(style.size());
+
+        self.local_scope_section
+            .content
+            .set_position_x(style.content.padding + style.content.size.x / 2.0);
+        self.local_scope_section
+            .content
+            .set_width(style.size_inner().x - 2.0 * style.content.padding);
+        self.local_scope_section.content.set_color(style.section.favourites_section_base_color);
+        self.local_scope_section.label.set_content(LOCAL_SCOPE_SECTION_HEADING_LABEL);
+
+        self.favourites_section.content.set_position_x(style.content.padding);
+        self.favourites_section.label.set_content(SUB_MODULES_SECTION_HEADING_LABEL);
+
+        self.sub_modules_section.content.set_position_x(style.content.padding);
+        self.sub_modules_section.label.set_content(SUB_MODULES_SECTION_HEADING_LABEL);
+
+        self.scroll_area.resize(Vector2::new(
+            style.content.size.x - style.content.padding,
+            style.content.size.y - style.content.padding,
+        ));
+        self.scroll_area.set_position_xy(Vector2::new(
+            -style.content.size.x / 2.0,
+            style.content.size.y / 2.0 - style.menu.height / 2.0,
+        ));
+        self.scroll_area.set_corner_radius_bottom_right(style.content.corner_radius);
+    }
+
+    fn recompute_layout(&self, style: &Style) {
+        self.update_style(style);
+        self.update_layers();
+
+        let favourites_section_height = self.favourites_section.height(style);
+        let local_scope_height = self.local_scope_section.height(style);
+        let sub_modules_height = self.sub_modules_section.height(style);
+
+        self.favourites_section.set_base_position_y(0.0, style);
+        self.local_scope_section.set_base_position_y(-favourites_section_height, style);
+        self.sub_modules_section
+            .set_base_position_y(-favourites_section_height - local_scope_height, style);
+
+        let full_height = favourites_section_height + local_scope_height + sub_modules_height;
+        self.scroll_area.set_content_height(full_height);
+        self.scroll_area.jump_to_y(full_height);
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "ComponentBrowserPanel"
+    }
+
+    fn new(app: &Application, _logger: &DefaultWarningLogger) -> Self {
+        Self::new(app)
+    }
+}
+
+
+
+// =======================
+// === Labeled Section ===
+// =======================
+
+/// Struct that contains the components contained in a section of the Component Browser Panel. Also
+/// provides some utility functions for shape and layout handling.
+#[derive(Clone, Debug, CloneRef)]
+struct LabeledSection<T: CloneRef> {
+    pub label:   text::Area,
+    pub divider: hline::View,
+    pub content: T,
+}
```

Comment:
Don't use constraints on structs! This makes these constraints appear everywhere. For example, do you really need `T` to be `CloneRef`able when invoking `set_style` function? Why? Why during the style setting `T` is required to be clonable? Basically, constraints on structs are considered a big mistake in Haskell and slowly in Rust as well. They have some use cases in Rust though - you need to use them (unfortunately) if Drop impl requires them. But hopefully we'll have a better way of doing that sooner than later. Anyway, don't use them until really necessary. Unlike other comments, this is a serious thing. If you'd have trouble fixing that, ping me and I'd love to help.

Reply:
This was the quickest solution to allow the struct to derive `CloneRef`, as the derive is not smart enough to handle the unconstrained type. I have now manually implemented `CloneRef` and removed the constraint here.